### PR TITLE
HPCC-28371 EclWatch Unable to spray CSV file due to "Invalid File Name"

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <Jeremy.Clements@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 1]
```diff
@@ -207,7 +213,7 @@ export const BlobImportForm: React.FunctionComponent<BlobImportFormProps> = ({
                                     rules={{
                                         required: nlsHPCC.ValidationErrorTargetNameRequired,
                                         pattern: {
-                                            value: /^(?:[\/\\]?[-a-z0-9_]+[\/\\])+([-a-z0-9 _\.]+)$/i,
+                                            value: /^(?:[\/\\]?[-a-z0-9_]+)+([-a-z0-9 _\.]+)$/i,
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:216]: The filename should not contain "\\" or "/"?

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 2]
```diff
@@ -227,7 +233,7 @@ export const DelimitedImportForm: React.FunctionComponent<DelimitedImportFormPro
                                     rules={{
                                         required: nlsHPCC.ValidationErrorTargetNameRequired,
                                         pattern: {
-                                            value: /^(?:[\/\\]?[-a-z0-9_]+[\/\\])+([-a-z0-9 _\.]+)$/i,
+                                            value: /^(?:[\/\\]?[-a-z0-9_]+)+([-a-z0-9 _\.]+)$/i,
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:236]: The filename should not contain "\\" or "/"?

File: esp/src/src-react/components/forms/landing-zone/JsonImportForm.tsx [hunk 3]
```diff
@@ -216,7 +222,7 @@ export const JsonImportForm: React.FunctionComponent<JsonImportFormProps> = ({
                                 rules={{
                                     required: nlsHPCC.ValidationErrorTargetNameRequired,
                                     pattern: {
-                                        value: /^(?:[\/\\]?[-a-z0-9_]+[\/\\])+([-a-z0-9 _\.]+)$/i,
+                                        value: /^(?:[\/\\]?[-a-z0-9_]+)+([-a-z0-9 _\.]+)$/i,
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/JsonImportForm.tsx:225]: The filename should not contain "\\" or "/"?

File: esp/src/src-react/components/forms/landing-zone/VariableImportForm.tsx [hunk 4]
```diff
@@ -210,7 +216,7 @@ export const VariableImportForm: React.FunctionComponent<VariableImportFormProps
                                     rules={{
                                         required: nlsHPCC.ValidationErrorTargetNameRequired,
                                         pattern: {
-                                            value: /^(?:[\/\\]?[-a-z0-9_]+[\/\\])+([-a-z0-9 _\.]+)$/i,
+                                            value: /^(?:[\/\\]?[-a-z0-9_]+)+([-a-z0-9 _\.]+)$/i,
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/VariableImportForm.tsx:219]: The filename should not contain "\\" or "/"?

File: esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx [hunk 5]
```diff
@@ -214,7 +220,7 @@ export const XmlImportForm: React.FunctionComponent<XmlImportFormProps> = ({
                                 rules={{
                                     required: nlsHPCC.ValidationErrorTargetNameRequired,
                                     pattern: {
-                                        value: /^(?:[\/\\]?[-a-z0-9_]+[\/\\])+([-a-z0-9 _\.]+)$/i,
+                                        value: /^(?:[\/\\]?[-a-z0-9_]+)+([-a-z0-9 _\.]+)$/i,
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx:223]: The filename should not contain "\\" or "/"?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx (hunk 1 above)
Comment (jeclrsg) [reply esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:216]: It should, but only here, as Dan pointed out in HPCC-28222. But I copied & pasted the regex into the other forms by mistake. Fixed in those.

Comment (GordonSmith) [reply esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:216]: @jeclrsg this filename could contain "\" or "/"?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx (hunk 1 above)
Comment (jeclrsg) [reply esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:216]: @GordonSmith No, I'd realized that second, non-optional directory separator there was incorrect. Testing various path-y type strings on https://regex101.com

**Review by GordonSmith — APPROVED** [review]
@jeclrsg can you rebase for 8.10.x?

**Pull request merged**